com.unity.perception/Runtime/Randomization/Samplers/ISampler.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using Unity.Collections;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace UnityEngine.Perception.Randomization.Samplers
+{
+    /// <summary>
+    /// Generates random values from probability distributions
+    /// </summary>
+    public interface ISampler
+    {
+        uint seed { get; set; }
+        FloatRange range { get; set; }
+
+        /// <summary>
+        /// Returns a duplicate sampler with an iterated seed
+        /// </summary>
+        /// <param name="index">
+        /// Offset value is often a the active scenario's currentIteration or framesSinceInitialization.
+        /// </param>
+        ISampler CopyAndIterate(int index);
```

Comment:
Samplers in parameters must have a way to transform their base state by some sort of index to support determinism during distributed execution. Since samplers mimic the behavior of the stateful NextSample() method used by Unity.Mathematics.Random, samplers attached to parameters cannot be directly sampled without modifying their internal state and breaking determinism. My solution was to offer convenience method for duplicating a generic sampler and iterating its base state based on whatever index is being used to enforce determinism across workloads.

Reply:
Update: samplers now have a baseSeed and a state, meaning its state can be mutated and reset at each scenario iteration. This new reset behavior is now executed for every parameter at the start of each scenario iteration and is transparent to the user. No more explicit state management for users. Yay!